lib/index.js

```diff
@@ -46,187 +43,101 @@ Ramen.prototype.initialize = function() {
-  var ramenSettings = {};
-  var property;
-  var identify;
-  var traits;
-  var opts;
-  var user_opts;
-  var created;
-  var name;
-  var id;
-  var email;
-  var group;
-
-  if (options.identify) {
-    identify = options.identify;
-    traits = identify.traits();
-    opts = identify.options(integration.name);
-    user_opts = opts.user;
-    created = identify.created();
-    name = identify.name();
-    id = identify.uid();
-    email = identify.email();
-    group = integration.analytics.group();
-  } else {
-    traits = {};
-    group = {};
-    opts = {};
+Ramen.prototype.page = function() {
+  if (!this.identified) {
+    window._ramen.identify();
+    this.identified = true;
   }
 
-  email = email || integration.analytics.user().traits().email;
-  id = id || integration.analytics.user().id();
+  window._ramen.page();
+};
 
-  // We need email addresses
-  if (!email) {
-    return;
+Ramen.prototype.track = function(track) {
+  if (!this.identified) {
+    window._ramen.identify();
+    this.identified = true;
   }
 
-  del(traits, 'email');
-  del(traits, 'name');
+  var properties = track.properties();
+  properties = convertDates(properties, function(date) { return Math.floor(date / 1000); });
+  window._ramen.track(track.event(), properties);
+};
 
-  if (traits.company !== null && !is.object(traits.company)) {
-    delete traits.company;
-  }
+/**
+ * Identify.
+ *
+ * @api public
+ * @param {Identify} identify
+ */
 
-  if (traits.company && typeof group.traits === 'function') {
-    defaults(traits.company, group.traits());
-  }
+Ramen.prototype.identify = function(identify) {
+  var user;
+  var traits;
+  var opts;
 
-  ramenSettings.organization_id = integration.options.organization_id;
-  ramenSettings.user = {
-    id: id,
-    email: email
-  };
+  traits = identify.traits();
+  opts = identify.options(this.name);
 
-  if (name) {
-    ramenSettings.user.name = name;
-  }
+  // Setup the basic `user` attributes: id, email, created_at, and name
+  // `null` values are OK. Ramen will ignore them.
 
-  if (created) {
-    ramenSettings.user.created_at = Math.round(created / 1000);
-    del(traits, 'created_at');
-    del(traits, 'created');
-    del(traits, 'createdAt');
-  }
+  user = { id: identify.userId() };
 
-  // Iterate through company traits and load them on ramenSettings.company
-  if (traits.company) {
-    var tc = traits.company;
-    ramenSettings.company = {
-      name: tc.name,
-      url: tc.url,
-      id: tc.id,
-      value: tc.value,
-      labels: tc.labels,
-      traits: {}
-    };
-
-    del(tc, 'name');
-    del(tc, 'url');
-    del(tc, 'id');
-    del(tc, 'value');
-    del(tc, 'labels');
-
-    tc = convertDates(tc, function(date) { return Math.floor(date / 1000); });
-    for (property in tc) {
-      if (tc.hasOwnProperty(property)) {
-        ramenSettings.company.traits[property] = tc[property];
-      }
-    }
-
-    del(traits, 'company');
+  if (!user.id) {
+    return;
```

Comment:
Ok cool then we're good to go!

I'll get working on updating our [docs page](https://gist.github.com/ryana/4bfe54d32ebcc229bba7) right now. Do I still just send that into partners@segment.com to get it updated?